### PR TITLE
Updated go to 1.23

### PR DESCRIPTION
**Pull request by shunr-hpe**
### Summary and Scope

Updated go to 1.23

### Issues and Related PRs

* Resolves CASMHMS-6317
 
### Testing

Tested on mug as part of sls

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,7 +24,7 @@
 # can't be run as these are just packages in this repo.
 
 # Build base has the packages installed that we need.
-FROM artifactory.algol60.net/docker.io/library/golang:1.16-alpine AS build-base
+FROM artifactory.algol60.net/docker.io/library/golang:1.23.3-alpine3.20 AS build-base
```
Comment (jwlv) [inline Dockerfile:27]: Would specifying ```artifactory.algol60.net/docker.io/library/golang:1.23-alpine``` here allow for auto-updates of any 1.23.x sub-version vs pinning to 1.23.3?  When I updated to Go 1.23 in other repos I was going down the path of 1.23 vs pinning to a specific 1.23 version.

File: Dockerfile.unittesting.Dockerfile [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 # Dockerfile for testing HMS s3 code.
 
-FROM artifactory.algol60.net/docker.io/library/golang:1.16-alpine AS build-base
+FROM artifactory.algol60.net/docker.io/library/golang:1.23.3-alpine3.20 AS build-base
```
Comment (jwlv) [inline Dockerfile.unittesting.Dockerfile:25]: Same question here for vs ```artifactory.algol60.net/docker.io/library/golang:1.23-alpine```

**Review by jwlv — APPROVED** [review]
My golang comments not a big deal, feel free to ignore.

File: Dockerfile (hunk 1 above)
Comment (shunr-hpe) [reply Dockerfile]: Thanks, I changed it to artifactory.algol60.net/docker.io/library/golang:1.23-alpine

**Review by mbuchmann-hpe — APPROVED** [review]

**Pull request merged**